### DISCUSSION TITLE
master分支5.15内核X86固件，云编绎报错。

### DISCUSSION THREAD

**Discussion by xyskwjy** [immortalwrt/immortalwrt#707] [Q&A]
ERROR: target/linux failed to build.
make -r world: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/workdir/openwrt/include/toplevel.mk:230: world] Error 1
make[2]: Entering directory '/workdir/openwrt/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/workdir/openwrt/scripts/config'
 make[1] world
 make[2] target/compile
 make[3] -C target/linux compile
    ERROR: target/linux failed to build.
make -r world: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/workdir/openwrt/include/toplevel.mk:230: world] Error 1
make[2]: Entering directory '/workdir/openwrt/scripts/config'
make[2]: 'conf' is up to date.
make[2]: Leaving directory '/workdir/openwrt/scripts/config'
make[1]: Entering directory '/workdir/openwrt'
make[2]: Entering directory '/workdir/openwrt'
make[3]: Entering directory '/workdir/openwrt/target/linux'
make[4]: Entering directory '/workdir/openwrt/target/linux/x86'
rm -f /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/localversion
/workdir/openwrt/scripts/kconfig.pl  + + /workdir/openwrt/target/linux/generic/config-5.15 /workdir/openwrt/target/linux/x86/config-5.15 /workdir/openwrt/target/linux/x86/64/config-5.15 > /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.target
awk '/^(#[[:space:]]+)?CONFIG_KERNEL/{sub("CONFIG_KERNEL_","CONFIG_");print}' /workdir/openwrt/.config >> /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.target
echo "# CONFIG_KALLSYMS_EXTRA_PASS is not set" >> /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.target
echo "# CONFIG_KALLSYMS_ALL is not set" >> /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.target
echo "CONFIG_KALLSYMS_UNCOMPRESSED=y" >> /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.target
/workdir/openwrt/scripts/package-metadata.pl kconfig /workdir/openwrt/tmp/.packageinfo /workdir/openwrt/.config 5.15 > /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.override
/workdir/openwrt/scripts/kconfig.pl 'm+' '+' /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.target /dev/null /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.override > /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set
mv /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.old
grep -v INITRAMFS /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.old > /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set
echo 'CONFIG_INITRAMFS_SOURCE=""' >> /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set
echo '# CONFIG_INITRAMFS_FORCE is not set' >> /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set
rm -rf /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/modules
cmp -s /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.prev || { cp /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config; cp /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.prev; }
export MAKEFLAGS= ; [ -d /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/user_headers ] || make -C /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40 KCFLAGS="-fmacro-prefix-map=/workdir/openwrt/build_dir/target-x86_64_musl=target-x86_64_musl" HOSTCFLAGS="-O2 -I/workdir/openwrt/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="x86_64-openwrt-linux-musl-" ARCH="x86" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Mon Jun  6 22:37:38 2022" KBUILD_BUILD_VERSION="0" KBUILD_HOSTLDFLAGS="-L/workdir/openwrt/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls=  KERNELRELEASE=5.15.40 CC="x86_64-openwrt-linux-musl-gcc" INSTALL_HDR_PATH=/workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/user_headers headers_install
grep '=[ym]' /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.config.set | LC_ALL=C sort | /workdir/openwrt/staging_dir/host/bin/mkhash md5 > /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.vermagic
touch /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.configured
rm -f /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/vmlinux /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/System.map
make -C /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40 KCFLAGS="-fmacro-prefix-map=/workdir/openwrt/build_dir/target-x86_64_musl=target-x86_64_musl" HOSTCFLAGS="-O2 -I/workdir/openwrt/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="x86_64-openwrt-linux-musl-" ARCH="x86" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Mon Jun  6 22:37:38 2022" KBUILD_BUILD_VERSION="0" KBUILD_HOSTLDFLAGS="-L/workdir/openwrt/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls=  KERNELRELEASE=5.15.40 CC="x86_64-openwrt-linux-musl-gcc" bzImage modules
make[5]: Entering directory '/workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40'
  SYNC    include/config/auto.conf.cmd
.config:1165:warning: override: CPU_FREQ_DEFAULT_GOV_SCHEDUTIL changes choice state
.config:3805:warning: override: M686 changes choice state
.config:7130:warning: override: SPARSEMEM_MANUAL changes choice state
*
* Restart config...
*
*
Error in reading or end of file.
* Processor type and features
make[7]: *** [scripts/kconfig/Makefile:77: syncconfig] Error 1
make[6]: *** [Makefile:616: syncconfig] Error 2
Makefile:706: include/config/auto.conf.cmd: No such file or directory
make[5]: *** [Makefile:729: include/config/auto.conf.cmd] Error 2
make[4]: *** [Makefile:29: /workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40/.modules] Error 2
make[3]: *** [Makefile:11: compile] Error 2
*
Symmetric multi-processing support (SMP) [Y/n/?] y
Processor feature human-readable names (X86_FEATURE_NAMES) [Y/?] y
make[2]: *** [target/Makefile:30: target/linux/compile] Error 1
make[1]: *** [target/Makefile:23: /workdir/openwrt/staging_dir/target-x86_64_musl/stamp/.target_compile] Error 2
Support x2apic (X86_X2APIC) [Y/?] y
Enable MPS table (X86_MPPARSE) [Y/n/?] y
Avoid speculative indirect branches in kernel (RETPOLINE) [Y/n/?] y
Mitigate Straight-Line-Speculation (SLS) [N/y/?] (NEW) 
make[5]: Leaving directory '/workdir/openwrt/build_dir/target-x86_64_musl/linux-x86_64/linux-5.15.40'
make[4]: Leaving directory '/workdir/openwrt/target/linux/x86'
make[3]: Leaving directory '/workdir/openwrt/target/linux'
time: target/linux/compile#1.29#0.32#1.57
    ERROR: target/linux failed to build.
make[2]: Leaving directory '/workdir/openwrt'
make[1]: Leaving directory '/workdir/openwrt'
make: *** [/workdir/openwrt/include/toplevel.mk:230: world] Error 2
Error: Process completed with exit code 2.

**Comment by 1715173329** [comment]
PR is welcome.
